proposals/network_canister_management/ledger_20211117T2319Z.md

```diff
@@ -0,0 +1,124 @@
+## Summary
+
+Updating Ledger to Wasm module with hash `61a312e4bf4f86899edabf6b3cedc504b2b0825d9b7956abea65582095692a9a` built from git commit `88d24e7def3b6ef41508d753f4037dd76d940a6d`.
+
+## Rationale
+
+This update contains changes required to allow all canisters to transfer ICP by sending messages to the Ledger canister:
+
+  * The ledger doesn't apply the whitelist when it decides whether to allow the transfer to the caller.
+    All principals except for the anonymous are allowed to transfer ICP now.
+
+  * The ledger exposes 2 new Candid endpoints: `account_balance` and `transfer` (see the full Candid interface below).
+    More Candid endpoints (e.g., an endpoint for fetching blocks) shall be added in subsequent proposals.
+
+The newly exposed candid interface:
+
+```candid
+// Amount of ICP tokens, measured in 10^-8 of a token.
+type ICP = record {
+     e8s : nat64;
+};
+
+// Number of nanoseconds from the UNIX epoch in UTC timezone.
+type Timestamp = record {
+    timestamp_nanos: nat64;
+};
+
+// AccountIdentifier is a 32-byte array.
+// The first 4 bytes is big-endian encoding of a CRC32 checksum of the last 28 bytes.
+type AccountIdentifier = blob;
+
+// Subaccount is an arbitrary 32-byte byte array.
+// Ledger uses subaccounts to compute the source address, which enables one
+// principal to control multiple ledger accounts.
+type SubAccount = blob;
+
+// Sequence number of a block produced by the ledger.
+type BlockIndex = nat64;
+
+// An arbitrary number associated with a transaction.
+// The caller can set it in a `transfer` call as a correlation identifier.
+type Memo = nat64;
+
+// Arguments for the `transfer` call.
+type TransferArgs = record {
+    // Transaction memo.
+    // See comments for the `Memo` type.
+    memo: Memo;
+    // The amount that the caller wants to transfer to the destination address.
+    amount: ICP;
+    // The amount that the caller pays for the transaction.
+    // Must be 10000 e8s.
+    fee: ICP;
+    // The subaccount from which the caller wants to transfer funds.
+    // If null, the ledger uses the default (all zeros) subaccount to compute the source address.
+    // See comments for the `SubAccount` type.
+    from_subaccount: opt SubAccount;
+    // The destination account.
+    // If the transfer is successful, the balance of this address increases by `amount`.
```

Comment:
The two lines above are a bit confusing. Generally: are we calling it account, account identifier or address?

Reply:
```suggestion
    // If the transfer is successful, the balance of this account increases by `amount`.
```